gamemodes/terrortown/gamemode/shared/sh_speed.lua

```diff
@@ -0,0 +1,100 @@
+local plymeta = assert(FindMetaTable("Player"), "FAILED TO FIND ENTITY TABLE")
+
+SPEED = SPEED or {}
+
+if SERVER then
+	function SPEED:HandleSpeedCalculation(ply, moveData)
+		if not ply:IsTerror() then return end
+
+		local baseMultiplier = 1
+		local isSlowed = false
+
+		-- Slow down ironsighters
+		local wep = ply:GetActiveWeapon()
+
+		if IsValid(wep) and wep.GetIronsights and wep:GetIronsights() then
+			baseMultiplier = 120 / 220
+			isSlowed = true
+		end
+
+		local speedMultiplierModifier = {1}
+		local returnMultiplier = hook.Run("TTTPlayerSpeedModifier", ply, isSlowed, moveData, speedMultiplierModifier) or 1
+
+		ply:SetSpeedMultiplier(baseMultiplier * returnMultiplier * speedMultiplierModifier[1])
+	end
+
+	---
+	-- A hook to modify the player speed, it is automatically networked.
+	-- @param Player ply The player whose speed should be modified
+	-- @param boolean isSlowed Is true if the player uses iron sights
+	-- @param CMoveData moveData The move data
+	-- @param table speedMultiplierModifier The speed modifier table. Modify the first table entry to change the player speed
+	-- @return [deprecated]number The deprecated way of changing the player speed
+	-- @hook
+	-- @realm server
+	function GM:TTTPlayerSpeedModifier(ply, isSlowed, moveData, speedMultiplierModifier)
+
+	end
+
+	---
+	-- Sets the speed multiplier to an absolute value
+	-- @param number value The new value
+	-- @realm server
+	function plymeta:SetSpeedMultiplier(value)
+		if self.lastSpeedMultiplier and self.lastSpeedMultiplier == value then return end
```

Comment:
is `self.lastSpeedMultiplier` (~= nil) really needed?

Reply:
probably not. However this will change because tkindanight noticed a bug in my implementation